pdf_parser/src/pdf_parser/pdf_extractor.py

```diff
@@ -0,0 +1,243 @@
+            for item in data:
+                if isinstance(item, (list, tuple)):
+                    file.write(", ".join(map(str, item)) + "\n")
+                else:
+                    file.write(str(item) + "\n")
+        else:
+            file.write(str(data))
+
+    
+
+prompt = """You are a JSON claim extraction specialist. Your task is to analyze input text and identify factual claims matching the following schema:
+        {{
+  "subject": "string",
+  "claim": "string",
+  "object": "string",
+  "statement": "string",
+  "aspect": "string",
+  "amt": 0,
+  "name": "string",
+  "howKnown": "FIRST_HAND",
+  "images": [
+    {{
+      "url": "string",
+      "metadata": {{
+        "captian": "string",
+        "description": "string"
+      }},
+      "effectiveDate": "2024-11-07T21:23:25.596Z",
+      "digestMultibase": "string",
+      "signature": "string",
+      "owner": "string"
+    }}
+  ],
+  "sourceURI": "string",
+  "effectiveDate": "2024-11-07T21:23:25.596Z",
+  "confidence": 0,
+  "claimAddress": "string",
+  "stars": 0
+}}
+
+
+        Meta Context for Claims:
+        
+claim: one of 'impact', 'rated', 'same_as'
+aspect: similar to 'impact:financial', 'impact:social', 'impact:work', 'quality:overall', 'quality:affordable', 'quality:fun', 'report:scam', 'report:dangerous'
+howKnown: will be set by the caller later
+effectiveDate: when the impact or rating occurred, if you can tell
+object: only if it has a url
+subject: try to find a url (may not be possible), otherwise use text
+statement: this is the concise narrative of the claim, quoted from the text
+amt: optional, generally only for impact claims and only if an amount is present.  The total impact claimed.
+unit: include if amt is set ; usd if dollars, or set according to what amt is counting
+stars: optional if claim is a rating, scale is 1-5, otherwise leave out of the result
+score: if claim is a rating, may be set from -1 to 1 as a float, otherwise unset.  5 stars = score of 1
+
+
+        Instructions:
+        1. Thoroughly examine the provided text while cross-referencing the schema requirements
+        2. Only extract claims that are explicitly stated or strongly implied in the text
+        3. Maintain strict adherence to the defined schema structure
+        4. If no claims match the criteria, return an empty array []
+        5. Never invent claims or use external knowledge
+        6. Prioritize precision over quantity
+
+        Output Guidelines:
+        - ALWAYS output a valid JSON array (starting with [ and ending with ])
+        - NEVER include markdown formatting or code blocks
+        - NEVER add explanations, disclaimers, or non-JSON content
+        - Ensure proper JSON syntax and escaping
+        - Maintain case sensitivity as defined in the schema
+
+        Response must be exclusively the JSON array with no additional text.
+
+"""
+API_KEY = os.getenv("ANTHROPIC_API_KEY")
+
+API_URL = "https://api.anthropic.com/v1/messages"
```

Comment:
this should be encapsulated, i thought it was?

Are you using messages bc its a thread, that's fine but put that in the extractor not in the pdf parser please